src/main/java/org/isf/patconsensus/service/PatientConsensusIoOperation.java

```diff
@@ -0,0 +1,106 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.isf.patconsensus.service;
+
+import java.util.Optional;
+
+import org.isf.opetype.model.OperationType;
+import org.isf.patconsensus.model.PatientConsensus;
+import org.isf.patient.service.PatientIoOperationRepository;
+import org.isf.utils.db.TranslateOHServiceException;
+import org.isf.utils.exception.OHServiceException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional(rollbackFor = OHServiceException.class)
+@TranslateOHServiceException
+public class PatientConsensusIoOperation {
+
+	@Autowired
+	private PatientConmsensusIoOperationRepository repository;
+
+	@Autowired
+	private PatientIoOperationRepository patientRepository;
+
+	/**
+	 * Return the PatientConsensus if exists {@link PatientConsensus}s
+	 *
+	 * @return the {@link PatientConsensus}s. It could be <code>empty</code>.
+	 * @throws OHServiceException
+	 */
+	public Optional<PatientConsensus> getPatientConsensusByUserId(Integer patientId) throws OHServiceException {
+		return repository.findByPatient_Code(patientId);
+	}
+
+	/**
+	 * Insert an {@link PatientConsensus} in the DB
+	 *
+	 * @param patientConsensus
+	 *            - the {@link PatientConsensus} to insert
+	 * @return <code>PatientConsensus</code> inserted.
+	 * @throws OHServiceException
+	 */
+	public PatientConsensus newPatientConsensus(PatientConsensus patientConsensus) throws OHServiceException {
+		patientConsensus.setPatient(this.patientRepository.findById(patientConsensus.getPatient().getCode()).get());
+		return repository.save(patientConsensus);
+	}
+
+	/**
+	 * Update an {@link PatientConsensus}
+	 *
+	 * @param patientConsensus
+	 *            - the {@link PatientConsensus} to update
+	 * @return <code>PatientConsensus</code>.
+	 * @throws OHServiceException
+	 */
+	public PatientConsensus updatePatientConsensus(PatientConsensus patientConsensus) throws OHServiceException {
+		patientConsensus.setPatient(this.patientRepository.findById(patientConsensus.getPatient().getCode()).get());
+		return repository.save(patientConsensus);
+	}
+
+	/**
+	 * Delete an {@link PatientConsensus}
+	 *
+	 * @param patientConsensus
+	 *            - the {@link PatientConsensus} to delete
+	 * @return <code>true</code> if the {@link PatientConsensus} has been delete, <code>false</code> otherwise.
+	 * @throws OHServiceException
+	 */
+	public void deletePatientConsensus(PatientConsensus patientConsensus) throws OHServiceException {
+		repository.delete(patientConsensus);
+	}
```

Comment:
this should not actually delete the record, a patient without related consensus record returns an error:

```
Exception in thread "AWT-EventQueue-0" java.util.NoSuchElementException: No value present
	at java.base/java.util.Optional.get(Unknown Source)
	at org.isf.patient.gui.PatientInsertExtended.getConsensus(PatientInsertExtended.java:2199)
	at org.isf.patient.gui.PatientInsertExtended.getJRightPanel(PatientInsertExtended.java:2190)
	at org.isf.patient.gui.PatientInsertExtended.getJDataPanel(PatientInsertExtended.java:391)
	at org.isf.patient.gui.PatientInsertExtended.getJContainPanel(PatientInsertExtended.java:375)
	at org.isf.patient.gui.PatientInsertExtended.initialize(PatientInsertExtended.java:354)
	at org.isf.patient.gui.PatientInsertExtended.<init>(PatientInsertExtended.java:338)
	at org.isf.admission.gui.AdmittedPatientBrowser.lambda$6(AdmittedPatientBrowser.java:829)
```

should it rather set all flags to "false"? in which case I would run a delete of a patient consensus record since patient itself cannot be hard deleted (but only soft deleted with `PAT_DELETED=Y|N`)